pkg/flare/archive.go

```diff
@@ -84,18 +87,25 @@ type filePermsInfo struct {
 	group string
 }
 
+// Profile is a performance profile containing heap and CPU profiles.
+type Profile struct {
+	FirstHeapProfile  []byte
+	CPUProfile        []byte
+	SecondHeapProfile []byte
+}
+
 // CreateArchive packages up the files
-func CreateArchive(local bool, distPath, pyChecksPath, logFilePath string) (string, error) {
+func CreateArchive(local bool, distPath, pyChecksPath, logFilePath string, profile *Profile) (string, error) {
 	zipFilePath := getArchivePath()
 	confSearchPaths := SearchPaths{
 		"":        config.Datadog.GetString("confd_path"),
 		"dist":    filepath.Join(distPath, "conf.d"),
 		"checksd": pyChecksPath,
 	}
-	return createArchive(zipFilePath, local, confSearchPaths, logFilePath)
+	return createArchive(confSearchPaths, local, zipFilePath, logFilePath, profile)
 }
 
-func createArchive(zipFilePath string, local bool, confSearchPaths SearchPaths, logFilePath string) (string, error) {
+func createArchive(confSearchPaths SearchPaths, local bool, zipFilePath, logFilePath string, profile *Profile) (string, error) {
```

Comment:
Why switch argument order ?

Reply:
Because then I only have to write `string` once. It's a minor change but I thought it's nicer.

Reply:
I agree that it's a nicer interface, it's an unexported function and the likelihood of a regression is small as we're covered by tests and the compiler would fail to build given the change. I'm fine with this.